mypy/nodes.py

```diff
@@ -194,6 +201,8 @@ class MypyFile(SymbolNode):
     path = ''
     # Top-level definitions and statements
     defs = None  # type: List[Statement]
+    # Type alias dependencies as mapping from node to set of alias full names
+    alias_deps = None  # type: DefaultDict[DepNode, Set[str]]
```

Comment:
These references should be merged after semantic analysis, or otherwise they can point to stale nodes.

Actually, is it necessary to have AST node references here? It looks like we could use the full name of the enclosing fine-grained target as the key instead, and this wouldn't have to be merged, and it would generally better match how things work in general). You could perhaps use `mypy.server.deps.Scope` to track the current target (it should be moved to a different module first since we don't want semantic analyzer to depend on anything in mypy.server).